src/pages/Browse/index.tsx

```diff
@@ -103,6 +106,10 @@ export class BrowseBase extends React.Component<Props> {
       return;
     }
 
+    if (version && currentVersionId !== version.id) {
```

Comment:
Switching from one `currentVersionId` to another needs a test case so that someone doesn't break this feature accidentally.